src/main/java/com/iexec/core/chain/adapter/UnauthenticatedBlockchainAdapterClient.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.core.chain.adapter;
+
+import com.iexec.common.config.PublicChainConfig;
+import feign.FeignException;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+
+/**
+ * Using an authenticated client makes the anonymous requests fail.
+ * So here is a client for anonymous requests.
+ */
+@FeignClient(
+        name = "UnauthenticatedBlockchainAdapterClient",
+        url = "#{blockchainAdapterClientConfig.url}"
+)
+public interface UnauthenticatedBlockchainAdapterClient {
```

Comment:
What about something like that to avoid multiple classes for same interface:
```
    @Bean
    public BasicAuthRequestInterceptor basicAuthRequestInterceptor() {
        return new BasicAuthRequestInterceptor(clientConfig.getUsername(),
                clientConfig.getPassword()) {

            @Override
            public void apply(RequestTemplate template) {
                if (Collections.singletonList("no-auth")
                        .equals(template.headers().get(AUTHORIZATION))) {
                    template.headers().remove(AUTHORIZATION);
                } else {
                    super.apply(template);
                }
            }
        };
    }
```
@Header("Authorization: 'no-auth' ")
``
?

Reply:
Well, that was useless. The fault was blockchain adapter's. It should have used `permitAll` method on '/config/chain' instead of 'anonymous'. It's fixed in [my last commit](https://github.com/iExecBlockchainComputing/iexec-blockchain-adapter-api/pull/23/commits/ab4816f820907ad2f1a9cb5280eae611e82c4445).

Reply:
Welldone :)